File: University/SeTayiFIsagurasi.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        long n = scanner.nextInt();

        for (long a = 1; a < n; a++) {
            for (long b = a; b < n; b++) {
                long c = n - a - b;
                if (a * a + b * b == c * c) {
                    ArrayList<Long> numbers = new ArrayList<>();
                    numbers.add(a);
                    numbers.add(b);
                    numbers.add(c);
                    Collections.sort(numbers);
                    System.out.println(numbers.get(0) + " " + numbers.get(1) + " " + numbers.get(2));
                    return;
                }
            }
        }
        System.out.println("Impossible");
    }
}

File: University/SeTayiFIsagurasi.py
def main():
    n = int(input("Enter a number: "))

    for a in range(1, n):
        for b in range(a, n):
            c = n - a - b
            if a * a + b * b == c * c:
                numbers = [a, b, c]
                numbers.sort()
                print(numbers[0], numbers[1], numbers[2])
                return

    print("Impossible")

if __name__ == "__main__":
    main()
